### PR TITLE
Replace local copyright check with pre-commit-hooks verify-copyright

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
The local `copyright.py` script is bug-prone. Replace it with a more robust centralized script from `pre-commit-hooks`.

Issue: https://github.com/rapidsai/build-planning/issues/30

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -107,6 +101,22 @@ repos:
         rev: v4.5.0
         hooks:
               - id: check-json
+      - repo: https://github.com/rapidsai/pre-commit-hooks
+        rev: v0.0.1
+        hooks:
+          - id: verify-copyright
+            files: |
+              (?x)
+                  [.](cmake|cpp|cu|cuh|h|hpp|sh|pxd|py|pyx)$|
```
Comment (benfred) [inline .pre-commit-config.yaml:110]: should we add in the rust files to this ?
```suggestion
                  [.](cmake|cpp|cu|cuh|h|hpp|sh|pxd|py|pyx|rs)$|
```

**Review by benfred — APPROVED** [review]
looks good - thanks!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (KyleFromNVIDIA) [reply .pre-commit-config.yaml:110]: That was not present in the old file, but I'll go ahead and add it.

**Review by raydouglass — APPROVED** [review]

**Pull request merged**